pkg/auth/providers/github/github_client.go

```diff
@@ -76,7 +76,27 @@ func (g *GClient) getUser(githubAccessToken string, config *v3.GithubConfig) (Ac
 		logrus.Errorf("Github getGithubUser: error unmarshalling response, err: %v", err)
 		return Account{}, err
 	}
-
+	// If the UserInfo endpoint does not return the email of a user, we need to query the emails endpoint
+	if githubAcct.Email == "" {
+		emailURL := g.getURL("USER_EMAIL", config)
+		emailResp, _, err := g.getFromGithub(githubAccessToken, emailURL)
+		if err != nil {
+			logrus.Errorf("Github getGithubUser: GET email url %v received error from github, err: %v", emailURL, err)
+			return Account{}, err
+		}
+		var userEmails []EmailResponse
+		if err := json.Unmarshal(emailResp, &userEmails); err != nil {
+			logrus.Errorf("Github getGithubUser: error unmarshalling email response, err: %v", err)
+			return Account{}, err
+		}
+		// This api endpoint doesn't accept query parameter to filter on primary, so we get all emails and loop through it
+		for _, userEmail := range userEmails {
+			if userEmail.Primary {
```

Comment:
Should we be also reading only Public?

Reply:
There can be only one primary email, and it's not necessary that's the one user marks as public. It could either be us storing the only primary email or the first public email. But I think either is fine since everything's returned by the github call response. I don't really have an opinion on which one to store, maybe @dnoland1 can tell?

Reply:
It's possible user has only one email saved and that is set to "private". So if we look for only the public one, we discard all emails that github api has returned to us. Plus on login page Github shows this to the user so user's aware of giving us read-only access to private email. Based on offline discussion with @deniseschannon and @will-chan we will store the primary one whether or not it's public
![new](https://user-images.githubusercontent.com/10672776/82619434-d1732d80-9b8a-11ea-92c5-51a0d9d19e8c.png)

Reply:
great, thanks